docs/references/config/config.toml.md

```diff
@@ -1584,79 +1584,53 @@ round of consensus will adopt increased timeout durations.
+How long a node waits, after receiving +2/3 conflicting prevotes/precommits, before pre-committing nil/going into a new round.
 
 ```toml
-timeout_prevote = "1s"
+timeout_round = "1s"
 ```
 
 | Value type          | string (duration) |
 |:--------------------|:------------------|
 | **Possible values** | &gt;= `"0s"`      |
 
+#### Prevotess
+
 A validator that receives +2/3 prevotes for a block, precommits that block.
 If it receives +2/3 prevotes for nil, it precommits nil.
 But if prevotes are received from +2/3 validators, but the prevotes do not
 match (e.g., they are for different blocks or for blocks and nil), the
-validator waits for `timeout_prevote` time before precommiting nil.
+validator waits for `timeout_round` time before precommiting nil.
 This gives the validator a chance to wait for additional prevotes and to
 possibly observe +2/3 prevotes for a block.
 
-Setting `timeout_prevote` to `0s` means that the validator will not wait
-for additional prevotes (other than the mandatory +2/3) before precommitting nil.
-This has important liveness implications and should be avoided.
-
-### consensus.timeout_prevote_delta
-
-How much the `timeout_prevote` increases with each round.
-
-```toml
-timeout_prevote_delta = "500ms"
-```
-
-| Value type          | string (duration) |
-|:--------------------|:------------------|
-| **Possible values** | &gt;= `"0ms"`     |
-
-Consensus timeouts are adaptive.
-This means that when a round of consensus fails to commit a block, the next
-round of consensus will adopt increased timeout durations.
-Timeouts increase linearly over rounds, so that the `timeout_prevote` adopted
-in round `r` is `timeout_prevote + r * timeout_prevote_delta`.
-
-### consensus.timeout_precommit
-
-How long a node waits, after receiving +2/3 conflicting precommits, before moving to the next round.
-
-```toml
-timeout_precommit = "1s"
-```
-
-| Value type          | string (duration) |
-|:--------------------|:------------------|
-| **Possible values** | &gt;= `"0s"`      |
+#### Precommits
 
 A node that receives +2/3 precommits for a block commits that block.
 This is a successful consensus round.
 If no block gathers +2/3 precommits, the node cannot commit.
 This is an unsuccessful consensus round and the node will start an additional
 round of consensus.
-Before starting the next round, the node waits for `timeout_precommit` time.
+Before starting the next round, the node waits for `timeout_round` time.
 This gives the node a chance to wait for additional precommits and to possibly
 observe +2/3 precommits for a block, which would allow the node to commit that
 block in the current round.
 
-Setting `timeout_precommit` to `0s` means that the validator will not wait
-for additional precommits (other than the mandatory +2/3) before moving to the
-next round.
-This has important liveness implications and should be avoided.
+#### Warning
+
+Setting `timeout_round` to `0s` means that the validator will not wait for
```

Comment:
We need a good way to minimize disruption for operators upgrading from current config (and not thinking of adapting this, so the new field will have value 0).
An idea can be that the new timeout, when 0, means some sensible default value (e.g. our default value: 1s).
Anyway, disabling these timeouts never made any sense to me in production. And if need to disable it in text, we can just set it to `time.Hour`.

Also, this design would enable us to (later on) add those timeouts as ConsensusParams:
* If the value is 0, the node uses the consensus param value
* But if the operator wants to override the consensus param value, they can by setting a value != 0 in the `config.toml`

Let me know what you think

Reply:
While this would work in this case, it won't in the case of `skip_timeout_commit`. If the user had `skip_timeout_commit=true` and `timeout_commit=1s`, in v1 if he/she doesn't do anything, the node will be running with `1s` timeout commit.

What if instead we version config and write a CLI tool for upgrading it? If the manual cmd is not desirable, then we can attempt upgrade upon a startup. How does that sound?

Reply:
Fair enough, but it risks being quite some effort. Maybe, a middle ground would be:
* We version the config as you say
* We include all changes to the config in a paragraph in `UPGRADING.md`
* Upon start-up, if the version received is
  * In the future, we panic saying this is an old version of CometBFT
  * In the past, we warn the user that they may have some fields that became obsolete
    * we include a link to the paragraph in `UPGRADING.md` to explain what to change
    * alternatively, for safely, we panic and ask the user to
      * bump the version manually
      * while doing that, to pay attention at the config fields that changed

If we decide to go the config tool path (I think it's overkill for the changes we have ATM), there is already one, called Confix, developed by the previous Tendermint Core team, so we should probably reuse it. It's on `v0.36.x` and `master`. AFAICT the SDK folks are also using it.

Reply:
Anyway, this discussion should block this PR (although the value 0 to `timeout_round` when upgrading isn't solved)

Reply:
@melekes FYI, I'm adding a point in the community call, to see what users think

Reply:
I will address this in a separate PR.